tiledb/sm/query/readers/reader_base.cc

```diff
@@ -1015,9 +1016,48 @@ Status ReaderBase::unfilter_tile_chunk_range(
     // Unfilter 't' for fixed-sized tiles, otherwise unfilter both 't' and
     // 't_var' for var-sized tiles.
     if (!var_size) {
-      if (!nullable)
-        RETURN_NOT_OK(unfilter_tile_chunk_range(
-            num_range_threads, range_thread_idx, name, t, tile_chunk_data));
+      if (!nullable) {
+        if (array_schema_.bitsort_filter_attr()) {
+          std::vector<Tile*> dim_tiles(array_schema_.dim_num());
+          std::vector<QueryBuffer> qb_vector;
+          std::vector<uint64_t> dim_data_sizes(array_schema_.dim_num());
+
+          for (size_t i = 0; i < array_schema_.dim_num(); ++i) {
+            const Dimension *dimension = array_schema_.domain().dimension_ptr(i);
+            auto dim_tile_tuple = tile->tile_tuple(dimension->name());
+            dim_tiles[i] = &dim_tile_tuple->fixed_tile();
+
+            auto &filtered_buffer = dim_tiles[i]->filtered_buffer();
+            dim_data_sizes[i] = filtered_buffer.size();
+            qb_vector.emplace_back(filtered_buffer.data(), nullptr, &dim_data_sizes[i], nullptr);
+          }
+
+          // Combine them into an argument.
```

Comment:
Is there any way we could share the code with the global order writer here?

Reply:
like how so? Add it to comparators.h?

Reply:
Nevermind I got it lol